config.yaml

```diff
@@ -0,0 +1,38 @@
+# Base URL to which clients will make their requests.
+# Useful if the API is proxied through reverse proxy like nginx.
+# Value needs to contain full URL with protocol scheme, e.g. https://braize.pajlada.com/chatterino
+#base-url: ""
+
+# Address to which API will bind and start listening on
+#bind-address: ":1234"
+
+# Max content size in bytes - requests with body bigger than this value will be skipped
+#max-content-length: 5242880
+
+# When enabled, will attempt to use lilliput library for building animated thumbnails.
+# Can increase memory usage by a lot.
+#enable-lilliput: true
+
+
+# Discord token, provides rich information for Discord invite links
+#discord-token: ""
+
+# Twitch client ID and Twitch client secret, provide rich information for Twitch Clips
+#twitch-client-id: ""
+#twitch-client-secret: ""
+
+# YouTube API key, provides rich information for YouTube video links
+#youtube-api-key: ""
+
+# Twitter bearer token, provides rich information for tweet and Twitter user links
+#twitter-bearer-token: ""
+
+# Imgur client ID, provides rich information for imgur image links
+#imgur-client-id: ""
+
+# oEmbed Facebook app ID and app secret, provide rich information for Facebook and Instagram links
+#oembed-facebook-app-id: ""
+#oembed-facebook-app-secret: ""
```

Comment:
It might be a good idea to remove the secret config values from this file. They should really be envvar only to prevent someone checking them in.

Reply:
This file is meant to show how default values can be configured by using a config file. Restricting secret values to be only configurable from the environment and cli flags misses the point of viper which should allow us to set any config value with any of three methods. I believe it is up to user to protect their credentials and prevent from any leaks ![zneixApu](https://static-cdn.jtvnw.net/emoticons/v2/306464464/default/dark/1.0) ![zneixLeak](https://static-cdn.jtvnw.net/emoticons/v2/306465249/default/dark/1.0)
@pajlada I hope my rambling makes sense and that you agree with me.

Reply:
I don't agree with limiting how people can configure the application - we provide 3 options: environment variables, command line options, and the file (from multiple places). You can configure the file in `/etc/chatterino-api/config.yaml` too